nibabel/testing/__init__.py

```diff
@@ -62,6 +65,22 @@ def assert_allclose_safely(a, b, match_nans=True, rtol=1e-5, atol=1e-8):
     assert_true(np.allclose(a, b, rtol=rtol, atol=atol))
 
 
+def check_iteration(iterable):
+    """ Checks that an object can be iterated through without errors. """
+    try:
+        for _ in iterable:
+            pass
+    except:
+        return False
+
+    return True
+
+
+def assert_arrays_equal(arrays1, arrays2):
+    for arr1, arr2 in zip_longest(arrays1, arrays2, fillvalue=None):
```

Comment:
Aha - thinking on this, I guess that these array sequences can be iterables.   In that case, could you document this, in the function?   Are these iterables that will be passed too large to convert them to lists and do `len` on them?  If so, maybe something like this to avoid the deprecation warning and to be explicit about the meaning of None:
```
for arr1, arr2 in zip_longest(arrays1, arrays2, fillvalue=None):
    assert_false(arr1 is None or arr2 is None)
    assert_array_equal(arr1, arr2)
```